src/components/button/Button.tsx

```diff
@@ -48,20 +54,21 @@ type ButtonIconProps = {
   notification?: ThemeIconProps['notification'];
 };
 
+type ButtonModeAwareProps =
+  | {mode?: 'primary'; interactiveColor?: InteractiveColor}
+  | {mode: Exclude<ButtonMode, 'primary'>; backgroundColor?: StaticColor};
+
 export type ButtonProps = {
   onPress(): void;
-  interactiveColor?: InteractiveColor;
-  mode?: ButtonMode;
-  viewContainerStyle?: StyleProp<ViewStyle>;
-  textContainerStyle?: StyleProp<ViewStyle>;
-  textStyle?: StyleProp<TextStyle>;
   leftIcon?: ButtonIconProps;
   rightIcon?: ButtonIconProps;
   active?: boolean;
   compact?: boolean;
   loading?: boolean;
   style?: StyleProp<ViewStyle>;
+  hasShadow?: boolean;
```

Comment:
Not any shadows on the component page in Figma. Ask christian if we still need it. Then it should be in Figma as well.

Or another option is wrapping the button with a view with shadow.

Reply:
@ckbilstad decided to not add in figma, so wrapped in a view

Reply:
Decided to go with this solution as it was the easiest way to solve it in regards to the button structure